includes/WcfceMailProcessor.inc

```diff
@@ -189,10 +193,201 @@ class WcfceMailProcessor {
+        'sequential' => 1,
+        'entity_table' => 'civicrm_activity',
+        'entity_id' => $activity_id,
+        'mime_type' => $file->filemime,
+        'name' => $file->filename,
+        'content' => $this->getWebformFile($file_id, 'content'),
+      ]);
+    }
+
+    return $result['values'][0] ?? [];
+  }
+
+  /**
+   * Encodes attached file contents to append them to email.
+   *
+   * @param string $headers
+   *   Email headers (used to get email boundary).
+   * @param array|null $message
+   *   (Optional) Email data. If set to NULL then $this->message would be used.
+   * @param string $crlf
+   *   (Optional) End of line character.
+   *
+   * @return string
+   *   Attached file contents (encoded), or empty string if no files found.
+   */
+  public function getWebformFilesAsEml($headers, $message = NULL, $crlf = MAIL_LINE_ENDINGS) {
+    $message = $message ?? $this->message;
+    $res = '';
+
+    // Check if files attached to a webform submission.
+    $file_ids = $this->getWebformFileIds($message);
+
+    if (empty($file_ids)) {
+      return $res;
+    }
+
+    $boundary = $crlf . '--' . $this->getEmailBoundary($headers) . $crlf;
+
+    // Attach files to email.
+    foreach ($file_ids as $file_id) {
+      $res .= $this->getWebformFile($file_id, 'content_eml', $crlf);
+      $res .= $boundary;
+    }
+
+    return $res;
+  }
+
+  /**
+   * Get ids of files attached to submitted webform.
+   *
+   * @param array|null $message
+   *   (Optional) Email data. If set to NULL then $this->message would be used.
+   *
+   * @return array
+   *   Ids of files attached to a webform submission, or empty array if no files
+   *   were attached.
+   *   The result array looks like: array(webform_field_num => file_id, ...).
+   */
+  public function getWebformFileIds($message = NULL) {
+    $message = $message ?? $this->message;
+    $fields = &$message['params']['node']->webform['components'];
+    $values = &$message['params']['submission']->data;
+    $res = [];
+
+    // Search for file type fields.
+    foreach ($fields as $num => $field) {
+      if ($field['type'] == 'file' && !empty($values[$num][0])) {
+        // Get id of uploaded file.
+        $res[$num] = $values[$num][0];
+      }
+    }
+
+    return $res;
+  }
+
+  /**
+   * Loads webform file.
+   *
+   * @param int $file_id
+   *   Drupal file id that is attached to a webform submission.
+   * @param string $return
+   *   (Optional) Type of the result to return. May be one of the following:
+   *   + object - (default) file object.
+   *   + content - file contents.
+   *   + content_eml - file contents encoded for eml.
+   * @param string $crlf
+   *   (Optional) End of line character, used for encoding to eml.
+   *
+   * @return string|object
+   *   File object or file contents depending on $return parameter.
+   */
+  public function getWebformFile($file_id, $return = 'object', $crlf = MAIL_LINE_ENDINGS) {
```

Comment:
I think we should have three functions `getWebformFileObject`, `getWebformFileContent` and `getWebformEmlContent` rather than a function that does so many things together.